packages/frontend/core/src/hooks/affine/use-members.ts

```diff
@@ -22,5 +24,47 @@ export function useMembers(
       take,
     },
   });
-  return data.workspace.members;
+
+  const members = data.workspace.members;
+
+  const sortedMembers = useMemo(() => {
+    // define a compare function
+    const compareMembers = (a: Member, b: Member) => {
```

Comment:
Calculate a weight value and order by it, not just scan properties to sort. Those have some issues when property values are the same, such as how to order when a and b all are Owner.

Reply:
```
members.sort((a, b) => {
  return calculateWeight(b) - calculateWeight(a)
})
```

Weight bigger first.

Reply:
thanks for your guidance